### ISSUE TITLE
Interest in pull request for PHPCS issues flagged by WordPress VIP?

### ISSUE THREAD

**Issue by davisshaver** [yikesinc/yikes-inc-easy-mailchimp-extender#848]
Hi there - We are looking at this plugin to enable Mailchimp integration on a site that is hosted on WordPress VIP Go. As part of the launch process, the code went through [automated PHPCS sniffing](https://wpvip.com/documentation/vip-go/vip-code-review/#automated-scans) by the WordPress VIP team. I have included a file showing output from the `WordPress-VIP-Go` ruleset. The majority of issues seem to be related to nonce use, escaping, and input sanitization.

Have plugin maintainers looked into this before? We could look into submitting a pull request (or pull requests) if this is of interest.

Thanks for the consideration.

[report.xlsx](https://github.com/yikesinc/yikes-inc-easy-mailchimp-extender/files/4599686/report.xlsx)

**Comment by TheTracyL** [comment]
Hello @davisshaver 

We are a VIP agency partner and were in the middle of going through the process of bringing the plugin up to VIP code standards until we got slammed with a big project. 

You can take a look at this branch in progress: https://github.com/yikesinc/yikes-inc-easy-mailchimp-extender/tree/add/vip-upgrade

If you would like to help, you can talk to @freddiemixell and coordinate. We would appreciate it!

-Tracy

**Comment by davisshaver** [comment]
@TheTracyL sounds good!

@freddiemixell, do you think this branch is able to be used at this point? Is there any specific area where you could use help?

**Comment by freddiemixell** [comment]
Hey @davisshaver!

I noticed you're in Philly as well that's pretty cool! To give you some context into this branch I've been following a methodology of WordPress plugin development started by [Alain Schlesser](https://www.alainschlesser.com/). You can see a great example of this style if you check out the way [Level Playing Field](https://github.com/yikesinc/yikes-inc-level-playing-field) plugin works.

My goal with this branch is to break the plugin apart into compartments and I've started with the shortcode.

Basically, you could pull down the branch and start working through bugs on the shortcode at this point. Once we can remove the shortcode files that will remove a ton of the phpcs errors.

This weekend I'm going to do some open source work on this to get re organized and come up with a list of tasks that will take this to the finish line!

Cheers,
Freddie

**Comment by davisshaver** [comment]
Thanks @freddiemixell! We'll watch for that. Sounds like this isn't being used in production yet, so we'll stick to the main branch for now and watch for an update from you later on.

**Comment by TheTracyL** [comment]
I'm closing this because it's in the works.
